netplan_cli/cli/commands/apply.py

```diff
@@ -290,11 +290,7 @@ def command_apply(self, run_generate=True, sync=False, exit_on_error=True, state
             # 2nd: start all other services
             utils.systemctl('start', netplan_wpa + netplan_ovs, sync=True)
         if restart_nm:
-            # Flush all IP addresses of NM managed interfaces, to avoid NM creating
-            # new, non netplan-* connection profiles, using the existing IPs.
             nm_interfaces = utils.nm_interfaces(restart_nm_glob, devices)
-            for iface in nm_interfaces:
-                utils.ip_addr_flush(iface)
```

Comment:
note: We need to confirm that this doesn't regress https://bugs.launchpad.net/netplan/+bug/1870561